packages/sdk-react/src/components/FuelProvider.tsx

```diff
@@ -0,0 +1,90 @@
+import type { Fuel } from '@fuel-wallet/sdk';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import type { ReactNode } from 'react';
+import { useContext, createContext, useEffect } from 'react';
+
+import { useWindowFuel } from '../hooks';
+import { QUERY_KEYS } from '../utils';
+
+export const fuelQueryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      // These two are annoying during development
+      retry: false,
+      refetchOnWindowFocus: false,
+      // This is disabled because it causes a bug with arrays with named keys
+      // For example, if a query returns: [BN, BN, a: BN, b: BN]
+      // with this option on it will be cached as: [BN, BN]
+      // and break our code
+      structuralSharing: false,
+    },
+  },
+});
+
+type FuelProviderProps = {
+  children?: ReactNode;
+};
+
+type FuelReactContextType = {
+  fuel: Fuel | undefined;
+};
+
+export const FuelReactContext = createContext<FuelReactContextType | null>(
+  null,
+);
+
+export const useFuel = () => {
+  return useContext(FuelReactContext) as FuelReactContextType;
+};
+
+export const FuelProvider = ({ children }: FuelProviderProps) => {
+  const fuel = useWindowFuel();
+
+  function onCurrentAccountChange() {
+    fuelQueryClient.fetchQuery([QUERY_KEYS.account]);
```

Comment:
Can you apply the changes I made here? https://github.com/FuelLabs/fuels-portal/blob/a2c20159bd6b040d53e62d43f02885b38ccb8556/packages/sdk-react/src/components/FuelProvider.tsx

Only the change  `fuelQueryClient.fetchQuery` to `fuelQueryClient.invalidateQueries`?